### DISCUSSION TITLE
Query returns null on similar type with different request variables

### DISCUSSION THREAD

**Discussion by Svarto** [urql-graphql/urql#2767] [Q&A]
Hi all, apologies if I am using the wrong terminology, I hope I can explain this as it is a headscratcher which I would really appreciate anyone's help untangling.

TLDR: When pulling one Query with a smaller subset of data, when navigating to another screen with a different Query that pulls the some data typename but with a different subset of requested data - the query returns data: null with no error and in Urql Devtools I get "undefined" for that typename.

Longer explanation:
I have Query 1 (see below for query), that is a infinite list with date as the cursor. When I pull this on a page all is fine. However, when I scroll down and pull the next data (i.e. update the cursor with the last items "created_at" number to move the cursor down the list) I get the data for Query 1, but if I navigate to another page with Query 2 (see below for query) it returns data: null. This happens even when I can confirm that the cursor argument for Query 2 is updated, so it is not the same as Query 1.

I receive no error, stale is false, and I just get back data: null. If I try to refresh the query by updating the cursor for Query 2 to `new Date()` (i.e. update it to NOW), then the same thing happens. See below for screenshot from Urql Devtools (note that the date format is different from React Native `new Date()`, which ends with Z, compared to the `created_at`, i.e. when the cursor is moved down the list to fetch a next batch of items, which ends with 00-00. Not sure if that is relevant):

![image](https://user-images.githubusercontent.com/24279217/198221999-fe2491b4-cbb6-46e6-87ec-a2795261be82.png)

Query 1:
```
// Used in collection bottom sheet
export const GET_OWN_RECIPES_WITH_COLLECTIONS = gql`
  query GetAllUserRecipes(
    $user_id: uuid!
    $cursorArgument: timestamptz
    $collection_id: Int
  ) {
    recipe(
      order_by: { created_at: desc }
      limit: 15
      where: {
        created_at: { _lt: $cursorArgument }
        user_id: { _eq: $user_id }
      }
    ) {
      dishtype
      created_at
      id
      user_id
      title
      draft
      images {
        id
        image_id
        user_id
      }
      collections_recipes(where: { collections_id: { _eq: $collection_id } }) {
        collections_id
        type
        recipe_id
      }
    }
  }
`;
```

Query 2:
```
// Used in my recipe feed
export const GET_USER_RECIPES = gql`
  query GetAllUserRecipes(
    $user_id: uuid!
    $cursorArgument: timestamptz
    $current_user: uuid
  ) {
    recipe(
      order_by: { created_at: desc }
      limit: 15
      where: {
        created_at: { _lt: $cursorArgument }
        user_id: { _eq: $user_id }
      }
    ) {
      dishtype
      created_at
      id
      user_id
      title
      draft
      images {
        image_id
        id
        user_id
      }
      recipe_likes_aggregate {
        aggregate {
          count
        }
      }
      recipe_likes(where: { user_id: { _eq: $current_user } }) {
        recipe_id
        user_id
      }
      cooked_recipes(where: { user_id: { _eq: $current_user } }) {
        id
      }
      cooked_recipes_aggregate {
        aggregate {
          count
        }
      }
    }
  }
`;
``` 

Happy to give more information, I am really scratching my head what is going wrong here...thanks for any help!!

**Comment by JoviDeCroock** [comment]
What exchanges are you using and what is the network response for query 2. With document cache both will be distinct fetches and with normalized cache probably as well since these require more data.

EDIT: in case you are using graphcache are you using pagination resolvers? Also are you seeing any warnings in the console?

**Reply by Svarto** [reply]
@JoviDeCroock thanks for the questions! I'm using the following exchanges:
- devtoolsExchange
- dedupExchange
- CacheExchange (graphcache)
- retryExchange
- fetchExchange
- subscriptionExchange

I'll check the network response time next time I sit by my development computer (tomorrow). Will get back to you soon!
